src/browserlib/extract-dfns.mjs

```diff
@@ -140,6 +206,18 @@ export default function (spec, idToHeading = {}) {
   }
 
   return [...document.querySelectorAll(definitionsSelector)]
+    .map(node => {
+      // 2021-06-21: Temporary preprocessing of invalid "idl" dfn type (used for
+      // internal slots) while fix for https://github.com/w3c/respec/issues/3644
+      // propagates to all EDs and /TR specs. To be dropped once crawls no
+      // longer produce warnings.
+      if (node.getAttribute('data-dfn-type') === 'idl') {
+        node.setAttribute('data-dfn-type', 'attribute');
```

Comment:
I've commented in https://github.com/w3c/respec/issues/3644#issuecomment-864894696 - I'm not convinced `attribute` is the right approach.

Code looks good otherwise.

Reply:
We're still discussing whether it would be worth creating an additional type in speced/spec-dfn-contract#1.

In the meantime, w3c/respec#3644 was closed with ReSpec now flagging internal slots with an `attribute` type and internal methods with a `method` type. I've updated the code to the same here and propose to go ahead with this while we converge on new types (or on making these dfns private by default one way or another).